test_launch_ros/test/test_launch_ros/actions/test_node.py

```diff
@@ -300,3 +299,33 @@ def test_launch_node_with_invalid_parameter_dicts(self):
                     },
                 },
             }])
+
+
+def test_node_name():
+    node_without_ns = launch_ros.actions.Node(
+        package='asd',
+        executable='bsd',
+        name='my_node',
+    )
+    lc = LaunchContext()
+    node_without_ns._perform_substitutions(lc)
+    assert not node_without_ns.is_node_name_fully_specified()
+
+    node_without_name = launch_ros.actions.Node(
+        package='asd',
+        executable='bsd',
+        namespace='my_ns',
+    )
+    lc = LaunchContext()
+    node_without_name._perform_substitutions(lc)
+    assert not node_without_name.is_node_name_fully_specified()
+
+    node_with_fqn = launch_ros.actions.Node(
+        package='asd',
+        executable='bsd',
+        name='my_node',
+        namespace='my_ns',
+    )
+    lc = LaunchContext()
+    node_with_fqn._perform_substitutions(lc)
+    assert node_with_fqn.is_node_name_fully_specified()
```

Comment:
I think it would make sense to add a test with this last check using `LifecycleNode` since that's the class changed by this PR.

The rest of the test make sense to me, though they don't seem to be related to this PR. I would approve regardless as long as a `LifecycleNode` test using `namespace` is added.

Reply:
Yeah, those test are a bit unrelated.
I should have added them in #153.

After https://github.com/ros2/launch_ros/pull/157/commits/a9e4c1045d3cd3c35b84021035a47598d69f5c14, the added tests are covering LifecycleNode too.

Reply:
It seems that I forgot to run tests locally before pushing :open_mouth:.
https://github.com/ros2/launch_ros/pull/157/commits/4253a34821937ca68f1d3d3d7a129bd60daa3060 should fix the problems.